api/dataset.go

```diff
@@ -2,38 +2,218 @@ package api
 
 import (
 	"context"
+	"errors"
 	"net/http"
+	"strings"
+
+	"github.com/ONSdigital/dp-api-clients-go/v2/cantabular"
+	"github.com/ONSdigital/dp-api-clients-go/v2/dataset"
+	"github.com/go-chi/chi/v5"
 )
 
-func (api *API) getDatasetJSON(w http.ResponseWriter, r *http.Request) {
-	var response *getDatasetJsonObservationsResponse
-	var err error
+type optionsMap map[string]map[string]dataset.Option
+
+type datasetParams struct {
+	id                 string
+	edition            string
+	version            string
+	basedOn            string
+	unsortedDimensions []string
+	options            optionsMap // dimension -> option -> option item
+}
 
+func (api *API) getDatasetJSON(w http.ResponseWriter, r *http.Request) {
 	ctx := r.Context()
 
 	if r.URL.Query().Get("geography") == "" {
-		response, err = api.getDefaultDatasetJSON(ctx, r)
+		api.getDefaultDatasetJSON(ctx, w, r)
 	} else {
-		response, err = api.getGeographyDatsetJSON(ctx, r)
+		api.getGeographyDatsetJSON(ctx, r)
+	}
+}
+
+func (api *API) getDefaultDatasetJSON(ctx context.Context, w http.ResponseWriter, r *http.Request) {
+	params, err := api.getDatasetParams(ctx, r)
+
+	if err != nil {
+		api.respond.Error(ctx, w, http.StatusNotFound, err)
```

Comment:
StatusNotFound? It seems this function can return a lot of errors that would be inappropriate to return a 404 for.
Also we have a convention of wrapping our errors with context at each level so the error messages are meaningful.
For example here we would do:
```
api.Respond.Error(ctx, w, statusCode(err), errors.Wrap(err, "failed to get dataset params" )
``` 
And then do the equivalent for all errors at all levels of the call stack. This avoids completely mysterious error messages like `invalid numeral at line 6` which might end up something like `failed to get dataset params: invalid dataset id: invalid numeral at line 6`